File: src/py/ShootNumber.py
OldCanopyShootNumber = CanopyShootNumber
EmergedLeaves = int(max(1, ceil(leafNumber - 1)))

Shoots = fibonacci(EmergedLeaves)

CanopyShootNumber = min(Shoots * SowingDensity, TargetFertileShoot)
AverageShootNumberPerPlant = CanopyShootNumber / SowingDensity
 
        
if (CanopyShootNumber != OldCanopyShootNumber):
    diff = CanopyShootNumber - OldCanopyShootNumber
    tilleringProfile.append(diff)
           
TillerNumber = len(tilleringProfile)
        
for i in range(len(leafTillerNumberArray),int(ceil(leafNumber))):
    leafTillerNumberArray.append(TillerNumber)
    

File: src/cpp/ShootNumber.cpp

double OldCanopyShootNumber = CanopyShootNumber;
int EmergedLeaves = (int)fmax(1, ceil(leafNumber - 1));
int Shoots = calculate_Fibonacci(EmergedLeaves).value;

CanopyShootNumber = fmin(Shoots * SowingDensity, TargetFertileShoot);
AverageShootNumberPerPlant = CanopyShootNumber / SowingDensity;

if (CanopyShootNumber != OldCanopyShootNumber)
{
	tilleringProfile.push_back(CanopyShootNumber - OldCanopyShootNumber);
}

TillerNumber = tilleringProfile.size();

for (int i = leafTillerNumberArray.size(); i < ceil(leafNumber); i++)
{
	leafTillerNumberArray.push_back(TillerNumber);
}
